### PR TITLE
Storybook vanilla modal

### PR DESCRIPTION
**Pull request by bobwise**
## What does this PR do?
Adds three variants of modal for the Vanilla Storybook site

### Associated Issue 
Fixes https://github.com/sparkdesignsystem/spark-design-system/issues/1809

### Code
 - [X] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [X] Unit Testing in Spark Vanilla with `npm run test` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [X] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### CODE REVIEW

File: spark/components/modals/react/SprkModal.stories.js [hunk 1]
```diff
@@ -58,7 +58,7 @@ export const waitStory = () => (
     isVisible={true}
     variant="wait"
   >
-    Wait Modal Content
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
```
Comment (afebbraro) [inline spark/components/modals/react/SprkModal.stories.js]: This line length is too long.

File: spark/components/modals/vanilla/modals.stories.js [hunk 2]
```diff
@@ -1,3 +1,216 @@
...
+export const info = () => {
+  useEffect(() => {
+    modals();
+  });
+
+  return `
+  <button
+  class="sprk-c-Button"
+  data-sprk-modal-trigger="exampleInfoModal"
+  type="button"
```
Comment (afebbraro) [inline spark/components/modals/vanilla/modals.stories.js]: Needs indentation.

File: spark/components/modals/vanilla/modals.stories.js [hunk 3]
```diff
@@ -1,3 +1,216 @@
...
+<div
+class="sprk-c-Modal sprk-u-Display--none"
+role="dialog"
+tabindex="-1"
+aria-labelledby="modalInfoHeading"
+aria-modal="true"
+aria-describedby="modalInfoContent"
+data-sprk-modal="exampleInfoModal"
+data-id="modal-info-1"
+>
```
Comment (afebbraro) [inline spark/components/modals/vanilla/modals.stories.js]: This needs some spacing/indentation.

File: spark/components/modals/vanilla/modals.stories.js [hunk 4]
```diff
@@ -1,3 +1,216 @@
...
+       </svg>
+     </button>
+   </header>
+
+   <div class="sprk-o-Stack__item sprk-c-Modal__body">
+     <p
+      class="sprk-b-TypeBodyTwo"
+      id="modalInfoContent"
+     >
+     Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
```
Comment (afebbraro) [inline spark/components/modals/vanilla/modals.stories.js]: This line length is too long.

File: spark/components/modals/vanilla/modals.stories.js [hunk 5]
```diff
@@ -1,3 +1,216 @@
...
+ class="sprk-c-Modal sprk-c-Modal--wait sprk-u-Display--none"
+  role="dialog"
+  tabindex="-1"
+  aria-labelledby="modalWaitHeading"
+  aria-modal="true"
+  aria-describedby="modalWaitContent"
+  data-sprk-modal="exampleWaitModal"
+  data-sprk-modal-type="wait"
+  data-id="modal-wait-1"
+ >
```
Comment (afebbraro) [inline spark/components/modals/vanilla/modals.stories.js]: This needs indentation/spacing adjustments.

File: spark/components/modals/vanilla/modals.stories.js [hunk 6]
```diff
@@ -1,3 +1,216 @@
...
+      const main = document.querySelector('[data-sprk-main]');
+
+      // Only proceed if this is a wait modal
+      if (!isWaitModal(modal)) return;
+
+      setTimeout(() => {
+        hideModal(modal, mask, main);
+      }, 5000);
+    });
+  });
```
Comment (RCopeland) [inline spark/components/modals/vanilla/modals.stories.js]: i dont think we need to have the whole close in 5 secs thing, since its wrapped in an iframe.

**Pull request merged**